### PR TITLE
Fix the '15 detector' limit on decomposing also applying when not decomposing errors

### PR DESCRIPTION
**Pull request by ChrisPattison**
Addresses #332

### CODE REVIEW

**Comment by ChrisPattison** [comment]
I'm pretty unhappy with the way in which I modified the test to pass. The error should really be some `X_ERROR` annotation with an exactly representable number like `0.125`, but the first check throws a decomposition failure instead of complaining about the number of syndromes.

File: src/stim/simulators/error_analyzer.test.cc [hunk 1]
```diff
@@ -2420,9 +2420,11 @@ TEST(ErrorAnalyzer, too_many_symptoms) {
     ASSERT_EQ("", check_catch<std::invalid_argument>("max supported number of symptoms", [&] {
                   ErrorAnalyzer::circuit_to_detector_error_model(symptoms_20, true, false, false, 0.0, false, true);
               }));
-    ASSERT_EQ("", check_catch<std::invalid_argument>("max supported number of symptoms", [&] {
-                  ErrorAnalyzer::circuit_to_detector_error_model(symptoms_20, false, false, false, 0.0, false, true);
-              }));
+    ASSERT_EQ(
```
Comment (Strilanc) [inline src/stim/simulators/error_analyzer.test.cc:2423]: Copy paste this case. It should pass when not decomposing (as you've done here) and fail if decomposing (which just requires setting one of the bool arguments to true).

**Review by Strilanc — CHANGES_REQUESTED** [review]

File: src/stim/simulators/error_analyzer.test.cc (hunk 1 above)
Comment (ChrisPattison) [reply src/stim/simulators/error_analyzer.test.cc:2423]: There was already a case to check for a failure when `decompose_errors = True`, so I just had to modify the test to accept the expected behavior when `decompose_errors = False`. Is that what you had in mind?

Comment (Strilanc) [reply src/stim/simulators/error_analyzer.test.cc:2423]: Ah, okay, then that's fine as is.

**Review by Strilanc — APPROVED** [review]

**Pull request merged**